src/skills-builder/skills-builder-modal/select-preferences/JobTitleSelect.jsx

```diff
@@ -12,16 +13,33 @@ import messages from './messages';
 
 const JobTitleSelect = () => {
   const { formatMessage } = useIntl();
-  const { dispatch, algolia } = useContext(SkillsBuilderContext);
+  const { state, dispatch, algolia } = useContext(SkillsBuilderContext);
   const { searchClient } = algolia;
+  const { currentJobTitle } = state;
 
   const handleCurrentJobTitleSelect = (value) => {
     dispatch(setCurrentJobTitle(value));
+    sendTrackEvent(
+      'edx.skills_builder.current_job.select',
+      {
+        category: 'skills_builder',
+        currentJob: value,
+      },
+    );
   };
 
-  // Below implementation sets the job title to "Student" or "Looking for work" — this overwrites any previous selection
-  // This will need to be revisited when we decide what to do with this data
-  const handleCheckboxChange = (e) => dispatch(setCurrentJobTitle(e.target.value));
+  const handleCheckboxChange = (e) => {
+    const { value } = e.target;
+    // only setCurrentJobTitle if the user hasn't selected a current job as we don't want to override their selection
+    if (!currentJobTitle) { dispatch(setCurrentJobTitle(value)); }
+
+    sendTrackEvent(
+      `edx.skills_builder.current_job.${value}`,
+      {
+        category: 'skills_builder',
+      },
+    );
+  };
```

Comment:
Potentially unrelated, but what if the user were to fill a checkbox _before_ searching for their `currentJob`? Would this conflict with any part of the user experience?

Reply:
Good question! So if they haven't set a `currentJob`, then this will dispatch an action (`setCurrentJobTitle`) to set it to `student` or `looking_for_work`. This will trigger the next question in the form.

If the user were to then set their job from the dropdown menu, that would overwrite `currentJob` to be whatever value they set it to.